cdap-hbase-compat-0.94/src/main/java/co/cask/cdap/data2/increment/hbase94/IncrementHandler.java

```diff
@@ -72,16 +83,50 @@
 
   private static final Log LOG = LogFactory.getLog(IncrementHandler.class);
 
+  private static final CompactionBound COMPACTION_BOUND_DEFAULT = CompactionBound.TX_UPPER_VISIBILITY_BOUND;
+
   private HRegion region;
   private TransactionStateCache cache;
 
+  protected Map<byte[], CompactionBound> compactionBoundByFamily = Maps.newTreeMap(Bytes.BYTES_COMPARATOR);
+
+  /**
+   * Defines compaction boundary for merging delta increments
+   */
+  public static enum CompactionBound {
```

Comment:
Do you envision another case beyond these two?  If not, just having a flag for whether or not transactions are enabled on the table or CF seems like it would be sufficient.